### PR TITLE
Add support for dispatching out of eth cores 

### PR DESCRIPTION
**Pull request by abhullar-tt**
- same mechanism as switching to FD out of eth as main (via `WH_ARCH_YAML` env flag)
- added terminate command called on device close 

https://github.com/tenstorrent/tt-metal/actions/runs/8723569411

### CODE REVIEW

File: .github/workflows/cpp-post-commit.yaml [hunk 1]
```diff
@@ -41,7 +41,7 @@ jobs:
         run: tar -xvf ttm_${{ matrix.runner-info.arch }}.tar
       - uses: ./.github/actions/install-python-deps
       - name: ${{ matrix.test-group.name }} tests
-        timeout-minutes: 15
+        timeout-minutes: 30
```
Comment (abhullar-tt) [inline .github/workflows/cpp-post-commit.yaml:44]: @tt-rkim we won't merge this to main, this is just temp to run some gtests using FD out of eth cores on CI

File: tests/tt_metal/tt_metal/unit_tests_common/compute/matmul/test_matmul_X_tile.cpp [hunk 2]
```diff
@@ -251,6 +251,9 @@ TEST_F(CommonFixture, MatmulSingleTile){
 }
 
 TEST_F(CommonFixture, MatmulMultiTile){
+    if (getenv("WH_ARCH_YAML")) {
```
Comment (abhullar-tt) [inline tests/tt_metal/tt_metal/unit_tests_common/compute/matmul/test_matmul_X_tile.cpp:254]: todo: remove this now

File: tests/tt_metal/tt_metal/unit_tests_common/dram/test_dram.cpp [hunk 3]
```diff
@@ -140,6 +140,9 @@ bool dram_single_core (CommonFixture* fixture, tt_metal::Device *device, const D
 }
 
 TEST_F(CommonFixture, DRAMLoopbackSingleCore){
+    if (getenv("WH_ARCH_YAML")) {
```
Comment (abhullar-tt) [inline tests/tt_metal/tt_metal/unit_tests_common/dram/test_dram.cpp:143]: todo: remove this

File: tt_metal/impl/device/device.cpp [hunk 4]
```diff
@@ -366,6 +358,14 @@ void Device::compile_command_queue_programs() {
                 CoreCoord completion_q_physical_core = get_physical_core_coordinate(completion_q_writer_location, dispatch_core_type);
                 CoreCoord dispatch_physical_core = get_physical_core_coordinate(dispatch_location, dispatch_core_type);
 
+                if (dispatch_core_type == CoreType::WORKER) {
```
Comment (abhullar-tt) [inline tt_metal/impl/device/device.cpp:361]: replace debug std out with log_debug

File: tt_metal/impl/device/device.cpp [hunk 5]
```diff
@@ -648,9 +666,9 @@ bool Device::close() {
 
     tt::Cluster::instance().set_internal_routing_info_for_ethernet_cores(false);
 
-    if (llrt::OptionsG.get_clear_l1()) {
-        this->clear_l1_state();
-    }
+    // if (llrt::OptionsG.get_clear_l1()) {
```
Comment (abhullar-tt) [inline tt_metal/impl/device/device.cpp:669]: add this back

File: tt_metal/impl/dispatch/command_queue_interface.hpp [hunk 6]
```diff
@@ -14,40 +14,90 @@
...
+        if (core_type == CoreType::WORKER) {
+            max_prefetch_command_size_ = 64 * 1024;
+            cmddat_q_size_ = 128 * 1024;
+            scratch_db_size_ = 128 * 1024;
+            dispatch_buffer_block_size = 512 * 1024;
+        } else {
+            max_prefetch_command_size_ = 32 * 1024;
+            cmddat_q_size_ = 64 * 1024;
+            scratch_db_size_ = 64 * 1024;
+            dispatch_buffer_block_size = 144 * 1024;
```
Comment (abhullar-tt) [inline tt_metal/impl/dispatch/command_queue_interface.hpp:83]: add back comment explaining why this has to be 2^N

**Review by pgkeller — APPROVED** [review]

File: tt_metal/impl/dispatch/kernels/cq_prefetch.cpp [hunk 7]
```diff
@@ -980,6 +980,10 @@ void kernel_main_h() {
             DPRINT << "terminating\n";
             done = true;
         }
+#if defined(COMPILE_FOR_IDLE_ERISC)
+        uint32_t heartbeat = 0;
+        RISC_POST_HEARTBEAT(heartbeat);
+#endif
```
[the remark this answers is not in the document]
Comment (abhullar-tt) [reply tt_metal/impl/dispatch/kernels/cq_prefetch.cpp:986]: will add heartbeat when prefetcher is stalled and in dispatcher is processing the wait

**Pull request merged**